compiler_opt/distributed/local/local_worker_manager.py

```diff
@@ -205,10 +209,37 @@ def shutdown(self):
       try:
         # Killing the process triggers observer exit, which triggers msg_pump
         # exit
+        self.resume()
         self._process.kill()
       except:  # pylint: disable=bare-except
         pass
 
+    def pause(self):
```

Comment:
I think you can have a test for pause/resume: the test worker keeps a list of timestamps. It adds every second to it. So you start it, wait 2 seconds, pause it, wait 5 seconds, resume, wait 2 seconds and read the list. There should be a >1 sec gap between 2 consecutive entries.